### PR TITLE
Add Helper Functions for Cleaning Data

### PR DESCRIPTION
**Pull request by dan-pomeranz**
Adds helper functions for cleaning a schema and tests for those functions

### CODE REVIEW

File: relationalize/schema.py [hunk 1]
```diff
@@ -150,6 +151,40 @@ def drop_null_columns(self) -> int:
...
 
+    def drop_special_char_columns(self) -> int:
+        """
+        Drops columns which have a non alnumeric (excluding whitespace) in their name from the schema.
+
+        Returns the # of columns that were dropped.
+        """
+        columns_to_drop = []
+        for key in self.schema.keys():
+            if re.sub(r"[^a-zA-Z0-9 ]", "", key) != key:
```
Comment (henryivesjones) [inline relationalize/schema.py:162]: this is a weird way to express what you want here I imagine find or match would be better

File: relationalize/schema.py [hunk 2]
```diff
@@ -150,6 +151,40 @@ def drop_null_columns(self) -> int:
...
+    def drop_duplicate_columns(self) -> int:
+        """
+        Drops columns from the schema which have a duplicate (case sensitive) match. Keeps the first column it reads.
+
+        Returns the # of columns that were dropped.
+        """
+        lowercased_keys = set()
+        columns_to_drop = []
+        for key in self.schema.keys():
+            print(key)
```
Comment (henryivesjones) [inline relationalize/schema.py:178]: dont print

File: test/schema.test.py [hunk 3]
```diff
@@ -196,6 +196,25 @@ def test_generate_output_columns_choice(self):
             schema1.generate_output_columns(),
         )
 
+    def test_drop_special_char_columns(self):
+        schema1 = Schema()
+        schema1.read_object({"abc ": 1, "def@#": 1, "$$ghi": 1, "jkl": 1, "!@#mno": 1})
+        schema1.drop_special_char_columns()
+        print(schema1.schema)
+        ##self.assertEqual(3, )
+        self.assertEqual(schema1.schema, {"abc ": "int", "jkl": "int"})
```
Comment (henryivesjones) [inline test/schema.test.py:205]: with unit testing usually you want to also test the null case (in this case this would be a schema with no columns that should be dropped)

File: test/schema.test.py [hunk 4]
```diff
@@ -196,6 +196,25 @@ def test_generate_output_columns_choice(self):
             schema1.generate_output_columns(),
         )
 
+    def test_drop_special_char_columns(self):
+        schema1 = Schema()
+        schema1.read_object({"abc ": 1, "def@#": 1, "$$ghi": 1, "jkl": 1, "!@#mno": 1})
+        schema1.drop_special_char_columns()
+        print(schema1.schema)
+        ##self.assertEqual(3, )
```
Comment (henryivesjones) [inline test/schema.test.py:204]: assert the return value

File: test/schema.test.py [hunk 5]
```diff
@@ -196,6 +196,25 @@ def test_generate_output_columns_choice(self):
             schema1.generate_output_columns(),
         )
 
+    def test_drop_special_char_columns(self):
+        schema1 = Schema()
+        schema1.read_object({"abc ": 1, "def@#": 1, "$$ghi": 1, "jkl": 1, "!@#mno": 1})
+        schema1.drop_special_char_columns()
+        print(schema1.schema)
```
Comment (henryivesjones) [inline test/schema.test.py:203]: dont print in tests

File: test/schema.test.py [hunk 6]
```diff
@@ -196,6 +196,25 @@ def test_generate_output_columns_choice(self):
...
+        schema1 = Schema()
+        schema1.read_object(
+            {"ABc ": 1, "DEf ": 1, "ghi": 1, "jkl": 1, "ABC": 1, "abc ": 1, "JkL": 1}
+        )
+        self.assertEqual(2, schema1.drop_duplicate_columns())
+        self.assertEqual(
+            schema1.schema,
+            {"ABc ": "int", "DEf ": "int", "ghi": "int", "jkl": "int", "ABC": "int"},
+        )
+
```
Comment (henryivesjones) [inline test/schema.test.py:217]: more cases (definitely include the noop case)

File: test/schema.test.py (hunk 4 above)
Comment (henryivesjones) [inline test/schema.test.py:204]: assert the return value

**Review by henryivesjones — COMMENTED** [review]
general idea is good

File: relationalize/schema.py (hunk 1 above)
Comment (dan-pomeranz) [reply relationalize/schema.py]: changed to a more readable equation

**Review by henryivesjones — APPROVED** [review]
lgtm
